File: Programming_2/C#/lab_4/Point.cs
namespace lab_4
{
    public class Point
    {
        public float X { get; set; }
        public float Y { get; set; }

        public Point():
            this(0, 0)
        {
        }
        public Point(float x, float y)
        {
            X = x;
            Y = y;
        }

        public override string ToString()
        {
            return $"{(X, Y)}";
        }
        
        public static Point operator +(Point point1, Point point2)
        {
            return new Point(point1.X + point2.X, point1.Y + point2.Y);
        }

        public static Point operator /(Point point, float num)
        {
            return new Point(point.X / num,
                            point.Y / num);
        }
        
        public static Point operator -(Point point1, Point point2)
        {
            return new Point(point1.X - point2.X, point1.Y - point2.Y);
        }
        
        public static Point operator *(Point point, float num)
        {
            return new Point(point.X * num,
                            point.Y * num);
        }
    }
}

File: Programming_2/C++/lab_5/lab_5.1/Point.cpp
//
// Created by Danil on 23.03.2020.
//

#include "Point.h"
#include <iostream>


Point::Point(): x(0), y(0){}

Point::Point(const Point &point): x(point.x), y(point.y) {}

Point::Point(float x, float y): x(x), y(y) {}

Point Point::operator+(const Point &other) const {
    return Point(x + other.x, y + other.y);
}

Point Point::operator-(const Point &other) const {
    return Point(x + other.x, y + other.y);
}

Point Point::operator*(float num) const {
    return Point(x * num, y * num);
}

Point Point::operator/(float num) const {
    return Point(x / num, y / num);
}


std::ostream& operator<<(std::ostream& stream,const Point& point) {
    stream << "(" << point.x << ", " << point.y << ")" << std::endl;
    return stream;
}

float Point::get_x() const {
    return x;
}

float Point::get_y() const {
    return y;
}

void Point::set_x(float value){
    x = value;
}

void Point::set_y(float value) {
    y = value;
}
